src/key_value_store.js

```diff
@@ -427,6 +427,20 @@ export class KeyValueStoreLocal {
     _getPath(fileName) {
         return path.resolve(this.localStoragePath, fileName);
     }
+
+    /**
+     * Retrieves the full public url of the designated file.
+     * @param {String} fileName 
+     */
+    _getPublicUrl( fileName ) {
+        ensureTokenOrLocalStorageEnvExists('key value store');
+        var storeId = process.env[ENV_VARS.DEFAULT_KEY_VALUE_STORE_ID]; 
+
+        const publicUrl = `https://api.apify.com/v2/key-value-stores/${storeId}/records/${fileName}`;
+        const publicLocalUrl = `file://${this.localStoragePath}/${fileName}`;
+
+        return process.env[_consts.ENV_VARS.LOCAL_STORAGE_DIR] && !this.forceCloud ? publicLocalUrl : publicUrl;
```

Comment:
We actually have two different classes that each represent either the remote key value store, that is reached via API, or the local one, which is stored in the file system. Instead of having this one function (that you placed in the `KeyValueStoreLocal` class), we should have two functions. One for the `KeyValueStore` and other for `KeyValueStoreLocal`. Each with their specific implementation.

Reply:
@mnmkng Alright so I pushed the requested changes. With the KeyValueStoreLocal vs KeyValueStore, I was thinking that this was the case, but I wasn't 100% sure, so I just put them together lol. Thanks for all the help and info!